### PR TITLE
feat(packages): get name of Homebrew tap if exists

### PR DESCRIPTION
**Pull request by mistydemeo**
This ensures that we preserve the full name of a package, tap included, if it was installed from a non-default tap. We do this by reading Homebrew's install receipt for packages we detected as coming from Homebrew. If the receipt exists, we read it out and check what tap the package is from; if it's not `homebrew/core`, we then prepend it to the detected package name.

### CODE REVIEW

File: cargo-dist/src/lib.rs [hunk 1]
```diff
@@ -966,7 +966,26 @@ impl Library {
...
+
+            // The path alone isn't enough to determine the tap the formula
+            // came from. If the install receipt exists, we can use it to
+            // get the name of the source tap.
+            let receipt = Utf8PathBuf::from(&prefix)
+                .join(&package)
+                .join("INSTALL_RECEIPT.json");
+
+            if receipt.exists() {
+                let _ = SourceFile::load_local(&receipt).and_then(|file| {
```
Comment (Gankra) [inline cargo-dist/src/lib.rs:979]: just checking we are in fact intentionally ignoring all errors and silently moving forward?

File: cargo-dist/src/lib.rs [hunk 2]
```diff
@@ -966,7 +966,26 @@ impl Library {
...
+                    file.deserialize_json().map(|parsed: serde_json::Value| {
```
Comment (Gankra) [inline cargo-dist/src/lib.rs:980]: idiomatically you should end the and_then after deserialize_json() and have the `map` chain after it (removes a level of indent)

File: cargo-dist/src/lib.rs [hunk 3]
```diff
@@ -966,7 +966,26 @@ impl Library {
...
+                        if let Some(tap) = parsed["source"]["tap"].as_str() {
+                            if tap != "homebrew/core" {
```
Comment (Gankra) [inline cargo-dist/src/lib.rs:982]: just checking, do we need to care about case here?

File: cargo-dist/src/lib.rs [hunk 4]
```diff
@@ -966,7 +966,26 @@ impl Library {
...
+            // came from. If the install receipt exists, we can use it to
+            // get the name of the source tap.
+            let receipt = Utf8PathBuf::from(&prefix)
+                .join(&package)
+                .join("INSTALL_RECEIPT.json");
+
+            if receipt.exists() {
+                let _ = SourceFile::load_local(&receipt).and_then(|file| {
+                    file.deserialize_json().map(|parsed: serde_json::Value| {
+                        if let Some(tap) = parsed["source"]["tap"].as_str() {
```
Comment (Gankra) [inline cargo-dist/src/lib.rs:981]: 👍 confirmed serde_json lets you do this and won't panic

**Review by Gankra — APPROVED** [review]
looks good, just some "does this work the way i think it does" questions

File: cargo-dist/src/lib.rs (hunk 2 above)
Comment (mistydemeo) [reply cargo-dist/src/lib.rs]: Oh cool, done.

File: cargo-dist/src/lib.rs (hunk 1 above)
Comment (mistydemeo) [reply cargo-dist/src/lib.rs]: Added a comment to that effect.

**Pull request merged**